File: scripts/parity/scenarios/js/spiral_simple.mjs
// Parity scenario: spiral using spiralXY geometry function
import { Point, spiralXY, transform } from '../../../../dist/index.js'

// Create a spiral using spiralXY
const centre_point = new Point({ x: 50, y: 50, z: 0.2 })
const start_radius = 10
const end_radius = 8
const start_angle = 0
const n_turns = 5
const segments = 320
const clockwise = true

const steps = spiralXY(centre_point, start_radius, end_radius, start_angle, n_turns, segments, clockwise)

const g = transform(steps, 'gcode', { show_tips: false, show_banner: false }).gcode
process.stdout.write(g.trimEnd() + '\n')

File: scripts/parity/scenarios/py/spiral_simple.py
import fullcontrol as fc
from math import tau

# Create a spiral using spiralXY
centre_point = fc.Point(x=50, y=50, z=0.2)
start_radius = 10
end_radius = 8
start_angle = 0
n_turns = 5
segments = 320
clockwise = True

steps = fc.spiralXY(centre_point, start_radius, end_radius, start_angle, n_turns, segments, clockwise)

res = fc.transform(steps, result_type='gcode')
print(res)
